src/Symfony/Component/Messenger/Transport/NullTransport.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Contracts\Service\ResetInterface;
+
+/**
+ * Transport that send nothing. Useful for testing purpose.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class NullTransport implements TransportInterface, ResetInterface
```

Comment:
Note implementing `ResetInterface` isn't enough for this to be properly integrated with the framework and stored envelopes to be cleared from one request to the next. The service definition must be tagged with `kernel.reset`.

But as the service is build using a factory and might use an env var for the DSN, it may not be possible to add the tag conditionally from the extension (compile time). Instead, you could implement ResetInterface on the factory too and keep track of created services to clear them from here (so only the factory will be tagged with `kernel.reset`).